lib/pharos/config.rb

```diff
@@ -66,12 +66,12 @@ def dns_replicas
 
     # @return [Array<Pharos::Configuration::Node>]
     def master_hosts
-      @master_hosts ||= hosts.select { |h| h.role == 'master' }
+      hosts.select { |h| h.role == 'master' }.sort_by(&:master_sort_score)
```

Comment:
```suggestion
      (@master_hosts ||= hosts.select { |h| h.role == 'master' }).sort_by(&:master_sort_score)
```

this would memoize the role selection part :)

Reply:
Too magical, hard to read 😄